lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +855,126 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+// \returns true if all dimensions of the input tensors are the same except
+// for the provided dimension, otherwise return false.
+static bool
+verifyInputsForConcat(const llvm::SmallVectorImpl<NodeValue> &inputs,
+                      size_t dimension) {
+  for (size_t i = 1; i < inputs.size(); i++) {
+    for (size_t j = 0; j < inputs[0].dims().size(); j++) {
+      if (j == dimension) {
+        continue;
+      }
+      if (inputs[0].dims()[j] != inputs[i].dims()[j]) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+// Given one tensor and the desired sub-tensor size, \returns the dimension,
+// after which the sub-tensor has the desired size, otherwise return -1.
```

Comment:
Can you give a small example in the doxygen comment? It's hard to visualize the sub-tensor and the large tensor without reading the code.

Reply:
added